File: java/ChatAPI/src/chat/api/logging/ILogger.java
package chat.api.logging;

public interface ILogger {
    void setDebugLogEnabled(boolean enabled);
    boolean isDebugLogEnabled();
    void setInfoLogEnabled(boolean enabled);
    boolean isInfoLogEnabled();
    void setWarnLogEnabled(boolean enabled);
    boolean isWarnLogEnabled();
    void setErrorLogEnabled(boolean enabled);
    boolean isErrorLogEnabled();

    void log(LogSeverity severity, String message);
    void logDebug(String message);
    void logInfo(String message);
    void logWarn(String message);
    void logWarn(Exception exception);
    void logError(String message);
    void logError(Exception exception);
}

File: c#/ChatAPI/Logging/ILogger.cs
namespace ChatAPI;

public interface ILogger
{ 
    bool DebugLogEnabled { get; set; }
    bool InfoLogEnabled { get; set; }
    bool WarnLogEnabled { get; set; }
    bool ErrorLogEnabled { get; set; }
    
    void Log(LogSeverity severity, string message);
    void LogDebug(string message);
    void LogInfo(string message);
    void LogWarn(string message);
    void LogWarn(Exception exception); 
    void LogError(string message);
    void LogError(Exception exception);
}